### DISCUSSION TITLE
How to debug post/put requests?

### DISCUSSION THREAD

**Discussion by jsjoeio** [johnlindquist/kit#64] [Q&A] [answered]
I'm running into issues using the `put` helper. I'm guessing I'm doing something wrong, but having a hard time debugging the 
request. In the logs, it says I'm getting a 403, but when I `curl` the request, it works just fine.

### `stop-toggl.js`

<details>
<summary>Click to toggle script contents</summary>

```js
let btoa = await npm("btoa")
let TOGGL_TOKEN = await env("TOGGL_TOKEN") //Go here -> https://track.toggl.com/profile

async function stopTimer(timeEntryId = "") {
  const url = `https://api.track.toggl.com/api/v8/time_entries/${timeEntryId}/stop`
  console.log('this is the url', url)
  console.log('this is the TOGGL TOKEN ', TOGGL_TOKEN)
  return await put(
    url,
    {
      headers: {
        "Content-Type": "application/json",
        Authorization:
          "Basic " + btoa(`${TOGGL_TOKEN}:api_token`),
      },
    }
  )
}

// In order to stop a time entry, we need a time_entry_id
const response = await get(
  "https://api.track.toggl.com/api/v8/time_entries/current",
  {
    headers: {
      "Content-Type": "application/json",
      Authorization:
        "Basic " + btoa(`${TOGGL_TOKEN}:api_token`),
    },
  }
)

const { data } = response.data
const timeEntryId = data.id
console.log(timeEntryId)

await stopTimer(timeEntryId)
```

</details>

## Steps to Reproduce
1. Set `TOGGL_TOKEN` 
2. Run script `stop-toggl`
3. See error 403 error `logs/console.log`

If you `curl` it, it should work:

```shell
curl -v -u 1971800d4d82861d8f2c1651fea4d212:api_token \
	-H "Content-Type: application/json" \
	-X PUT https://api.track.toggl.com/api/v8/time_entries/436694100/stop
```

Reference: [Toggl Docs](https://github.com/toggl/toggl_api_docs/blob/master/chapters/time_entries.md#stop-a-time-entry)

**Comment by johnlindquist** [comment, marked as answer]
When you reference `put`, it pulls in "axios":

https://github.com/johnlindquist/kit/blob/main/preload/api.cjs#L21

Here's the docs:

https://github.com/axios/axios#axiosputurl-data-config
